src/components/layouts/Header.jsx

```diff
@@ -1,33 +1,54 @@
-import { Link, useLocation, useNavigate } from "react-router-dom";
-import { useEffect, useState } from "react";
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+import { useEffect, useState } from 'react';
 
-import DesktopNav from "../navbars/DesktopNav";
-import MobileNav from "../navbars/MobileNav";
+import DesktopNav from '../navbars/DesktopNav';
+import MobileNav from '../navbars/MobileNav';
 
 const Header = () => {
-    const [isSearch, setIsSearch] = useState(true);
-    const [isLoginBtn, setIsLoginBtn] = useState(true);
-    const [isJoinBtn, setIsJoinBtn] = useState(true);
-    
-    const location = useLocation();
-    useEffect(() => {
-        location.pathname === "/search" ? setIsSearch(false) : setIsSearch(true);
-        location.pathname === "/login" ? setIsLoginBtn(false) : setIsLoginBtn(true);
-        location.pathname === "/register" ? setIsJoinBtn(false) : setIsJoinBtn(true);
-    }, [location.pathname]);
+  const [isSearch, setIsSearch] = useState(true);
+  const [isLoginBtn, setIsLoginBtn] = useState(true);
+  const [isJoinBtn, setIsJoinBtn] = useState(true);
 
-    const navigate = useNavigate();
-    const search = () => navigate("/search");
-    const login = () => navigate("/login");
-    const register = () => navigate("/register");
+  const location = useLocation();
+  useEffect(() => {
+    location.pathname === '/search' ? setIsSearch(false) : setIsSearch(true);
+    location.pathname === '/login' ? setIsLoginBtn(false) : setIsLoginBtn(true);
+    location.pathname === '/register'
+      ? setIsJoinBtn(false)
+      : setIsJoinBtn(true);
+  }, [location.pathname]);
 
-    return (
-        <header className="h-20 w-full flex items-center px-4 lg:px-20 justify-between">
-            <Link className="text-black font-spartan font-semibold text-xl sm:text-3xl w-1/3" to="/">TalentStudio</Link>
-            <DesktopNav onSearch={search} onLogin={login} onJoin={register} isSearch={isSearch} isLogin={isLoginBtn} isJoin={isJoinBtn}/>
-            <MobileNav onSearch={search} onLogin={login} onJoin={register} isSearch={isSearch} isLogin={isLoginBtn} isJoin={isJoinBtn}/>
-        </header>
-    );
+  const navigate = useNavigate();
+  const search = () => navigate('/search');
```

Comment:
This logic just redirect user on the search page, but does not save search request. Tat's why the user need to enter search request again after redirection on the separate search bar. It is not use friendly. I think you have at least 2 approaches:
1. show search results after redirection without printing them again in a separate input
2. get rid of search input in the Navbar if this input does not handle search requests.

Please think about what to do in this case and make sure that you have ticket for this in the Jira

Reply:
I think Tanya needs to add a state for search input in the header, store the data via onChange and then when the user click on the button or press the key Enter, the state will be passed as the initial state of the search input on the Search page.

Reply:
> I think Tanya needs to add a state for search input in the header, store the data via onChange and then when the user click on the button or press the key Enter, the state will be passed as to the search input on the Search page.

We discussed it with Tanya and decided that it is the next ticket.